Patch origin: in-house
Patch status: Solaris-specific; not suitable for upstream

Redirects build process to use local archives instead of downloading
from external webs.

--- apache-tomcat-8.5.83-src/build.properties.default
+++ apache-tomcat-8.5.83-src/build.properties.default
@@ -121,6 +121,12 @@ base-maven.loc=https://repo.maven.apache
 # Mirror, was used when there were problems with the main SF downloads site
 # base-sf.loc=https://sunet.dl.sourceforge.net
 
+# In Userland we really don't want Tomcat to download its build
+# dependencies itself but instead we will do it ourself in advance.
+# Therefore those dependencies which are really required need to be
+# redirected to local file system.
+userland-files.loc=file://${base.path}/../../
+
 # ----- Webservices - JAX RPC -----
 jaxrpc-lib.version=1.1-rc4
 jaxrpc-lib.checksum.enabled=true
@@ -151,13 +157,13 @@ # http://archive.eclipse.org/eclipse/dow
 #
 jdt.version=4.6.3
 jdt.release=R-4.6.3-201703010400
-jdt.checksum.enabled=true
+jdt.checksum.enabled=false
 jdt.checksum.algorithm=SHA-512
 jdt.checksum.value=372da97d5f37095a3616c730170dc541b93acb8f50370cd9b0d81fba4583e865f0c933dc012155f26638306b82a402eaab209123cd0a979fa7c86313c0fca477
 jdt.home=${base.path}/ecj-${jdt.version}
 jdt.jar=${jdt.home}/ecj-${jdt.version}.jar
 # The download will be moved to the archive area eventually. We are taking care of that in advance.
-jdt.loc.1=http://archive.eclipse.org/eclipse/downloads/drops4/${jdt.release}/ecj-${jdt.version}.jar
+jdt.loc.1=file://${base.path}/ECJ/ecj.jar
 jdt.loc.2=http://download.eclipse.org/eclipse/downloads/drops4/${jdt.release}/ecj-${jdt.version}.jar
 
 # ----- Tomcat native library -----
@@ -311,3 +317,11 @@ jsign.checksum.value=10fb38a1182515d583a
 jsign.home=${base.path}/jsign-${jsign.version}
 jsign.jar=${jsign.home}/jsign-${jsign.version}.jar
 jsign.loc=${base-maven.loc}/net/jsign/jsign/${jsign.version}/jsign-${jsign.version}.jar
+
+# allow Tomcat unit test pass even with JDK 8
+# https://bz.apache.org/bugzilla/show_bug.cgi?id=66064
+opens.javalang=-Dnop
+opens.javaio=-Dnop
+opens.sunrmi=-Dnop
+opens.javautil=-Dnop
+opens.javautilconcurrent=-Dnop
